midas/run/species.py

```diff
@@ -140,7 +140,10 @@ def read_gene_lengths(args, species_info, marker_info):
 	""" Read in total gene length per species_id """
 	total_gene_length = dict([(_,0) for _ in species_info])
```

Comment:
total_gene_length = defaultdict(int)

Altho the existing version has the benefit of crashing if you encounter a key not in species_info.